### PR TITLE
Follow up mqtt broker ux

### PR DESCRIPTION
**Pull request by cstns**
## Description

general code cleanup, e2e improvements

 - [x] Extract standalone logic into distinct components
 - [x] improve css styling/rules
 - [x] added new icon for broker empty state
 - [x] addressed snapshot.spec missing cypress selectors
 - [x] add more and improve existing ee e2e tests
 - [x]  enhanced form validation when editing/creating/altering broker client and acls

## Related Issue(s)

closes https://github.com/FlowFuse/flowfuse/issues/4685
closes https://github.com/FlowFuse/flowfuse/issues/4702

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/ui-components/stylesheets/ff-utility.scss [hunk 1]
```diff
@@ -21,4 +21,10 @@ $ff-funit-xs: 0.75rem;
...
 $ff-funit-lg: 1.5rem;
-$ff-funit-xl: 2rem;
\ No newline at end of file
+$ff-funit-xl: 2rem;
+
+@mixin ellipsis {
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
```
Comment (Steve-Mcl) [inline frontend/src/ui-components/stylesheets/ff-utility.scss:30]: We already have the class `truncate` that is exactly the same as this new `ellipsis`

image

**Review by Steve-Mcl — COMMENTED** [review]
Can we use Tertiary buttons for `+` and `-` & check W/H are equal & H + V alignment please?

image

File: frontend/src/ui-components/stylesheets/ff-utility.scss (hunk 1 above)
Comment (cstns) [reply frontend/src/ui-components/stylesheets/ff-utility.scss:30]: after looking into it a bit more after our conversation, Tailwind CSS does not use SCSS or mixins so i don't have the possibility to use the .truncate class as a mixin like i'm using it in this situation. In order to preserve annotation with tailwind, I renamed the mixin to follow suite tailwind's nomenclature.

Another solution would be to simply remove the mixin altogether and use the css rules as is, but I found that i'm using these classes quite extensively lately and would appreciate a shorthand for css imports where i'm unable to use tailwind's helper classes

**Comment by Steve-Mcl** [comment]
@cstns 

We seem to have lost the copy confirmation.

Before and after
chrome_2M8iUiLTxd

**Review by Steve-Mcl — APPROVED** [review]
Nice :)

**Pull request merged**